File: 100doors.py
#A solution to the 100 doors problem

doors = [0] * 100
    
def toggle(n):
    if(n == 101):
        return
    for i in range(100):
        if (i + 1) % n == 0:
            doors[i] = 0 if doors[i] else 1
    toggle(n + 1)
    
toggle(1)

[print('' if door == 0 else index + 1, end=' ' if door == 1 else '') for index, door in enumerate(doors)]

File: 100doors.js
// A node js solution to the 100 doors problem#A solution to the 100 doors problem

const opened = (function doors (door, n) {
    if(n === 101)
        return door;
    for(let i = n; i < 100; i++)
        if ((i + 1) % n === 0)
            door[i] = door[i] === 1 ? 0 : 1;
    return doors(door, n + 1);
}([...new Array(100)].map(() => 1), 1))

opened.forEach((element, index) => { if(element === 1 ) process.stdout.write(index + 1 + " "); })